src/hotspot/share/gc/parallel/psPromotionLAB.hpp

```diff
@@ -118,9 +118,9 @@ class PSOldPromotionLAB : public PSPromotionLAB {
     // assert(_state != flushed, "Sanity");
     assert(_start_array != NULL, "Sanity");
     HeapWord* obj = top();
-    HeapWord* new_top = obj + size;
-    // The 'new_top>obj' check is needed to detect overflow of obj+size.
-    if (new_top > obj && new_top <= end()) {
+    // Pointer overflow check is needed here.
```

Comment:
The comment mentions pointer overflow checking, but that's not what's
being checked for anymore.  It's just checking whether there is space
for the desired object.  It's just that the check is now being done
safely and correctly, where it wasn't before, but that's not interesting.
I don't think any comment is needed anymore; the updated code (with
my other suggestions) is obvious.

Reply:
Will remove in next revision. Thanks.